### PR TITLE
validate stable rc review process

### PR DESCRIPTION
**Pull request by nareshkamboju**
Stable-rc review validation process document is created.
Signed-off-by: Naresh Kamboju <naresh.kamboju@linaro.org>

### CODE REVIEW

**Comment by nareshkamboju** [comment]
Hi Team,

Please review this document and provide early commits for corrections.
Ref:
Stable review process not documented #20 
https://github.com/Linaro/lkft-tools/issues/20

File: docs/validate-stable-rc-builds.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
+
+# Stable rc review validation process
+Following steps to be performed when a stable rc review pushed on stable mailing list
+
+## Wait for stable rc review email
+Greg KH pushes on to stable rc trees and creates a tag
+then sends out email on stable ML with the list of patches included in that
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: Can you be more specific here about the process? Perhaps link "stable rc trees" to the actual repo. Same with mailing list.

File: docs/validate-stable-rc-builds.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+branch.
+
+You may check your inbox with give filters to identify the stable rc review
+for a given branch.
+
+## Build Test
+### Check for build is PASS
+open lkft.linaro.org website and check the build status
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md:32]: link

File: docs/validate-stable-rc-builds.md [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+From lkft.linaro.org you will re-directed to each branch builds for
+each device. If you see build status is in progress wait for the
+build to finish.
+
+https://ci.linaro.org/job/openembedded-lkft-linux-stable-rc-4.9/
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: why is this here?

File: docs/validate-stable-rc-builds.md [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+
+### Report build failure
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md:39]: Perhaps an example would help?

File: docs/validate-stable-rc-builds.md [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+
+### Bisecting bad commit
+git bisect is helpful for finding the bad commit
+If you could track down and find bad commit causing build failure then
+report the detailed email on stable rc ML and include patch author and
+Sub-system maintainers.
+
+### Cancel multiple pushes
+If you see multiple pushes in short period of time on a given branch.
+Cancel the old build jobs on ci.linaro.org
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: How? Link?

File: docs/validate-stable-rc-builds.md [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+## LAB ready
+Check all devices status on lkft.linaro.org instance,
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md:84]: What is 'lkft.linaro.org instance'?

File: docs/validate-stable-rc-builds.md [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+make sure you have good number of devices are online to finish the testing
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md:85]: What is a good number?

File: docs/validate-stable-rc-builds.md [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+in time. If devices are offline check with LAB team and know the reason for
+offline or maintenance mode
+
+Ping LAB team on IRC and report device related problem.
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md:89]: Where on IRC?

File: docs/validate-stable-rc-builds.md [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+Mouting NFS rootfs server might be a problem.
+Serial connection problem.
+LAVA dispatcher problem.
+
+### Sanity test
+For each branch build testing the sanity test will be done and these
+sanity jobs have high priority and runs LTP ltp-quickhit-tests jobs on
+4 architecture's devices arm64, arm, x86_64 and i386.
+
+Check dmesg and ensure no kernel panic, BUG, warning in boot log and test log.
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: How do you actually do this?

File: docs/validate-stable-rc-builds.md [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+If you find Kernel panic or warnings please report on stable ML
+If you find boot failure with or without kernel crash log please report on
+stable ML.
+Check each test results and ensure no failures
+If any find any test failure, try to reproduce the test failure with possible
+minimal steps and report failure on stable ML include respective mailing lists
+for example LTP / kselfest / netdev / bpf mailing lists 
+
+https://qa-reports.linaro.org/lkft/linux-stable-rc-4.9-oe-sanity
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: What is this for?

File: docs/validate-stable-rc-builds.md [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+
+### Validate each branch results
+Each stable rc branch have defined LAVA test job priority.
+Please refer doc/lava-job-priority.md file for more details.
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: link this

File: docs/validate-stable-rc-builds.md [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+
+The 4.9 branch have high priority so check 4.9 branch jobs are getting
+finished first and look for the test finished count on
+
+https://qa-reports.linaro.org/lkft/linux-stable-rc-4.9-oe/
+
+The each device submits 16 common jobs and few special test suites
+
+#### Number of tests
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md:138]: For this section and the next two, I'm afraid they'll go out of date fast and there may be a better place to refer to the data. (I'm not certain where, though we do have some on lkft.linaro.org.

File: docs/validate-stable-rc-builds.md [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+Remove the list of patch commit headers and file details
+Copy and paste results summary generated by lkft-tools generate_lts_report.py
+
+Ex:
+ - Reply all to stable-rc 4.9 e-mail
+ - python3 generate_lts_report.py 4.9 > 4.9.results.file
+open and review 4.9.results.file
+do the required changes as described above
+copy and paste results summary on stable-rc 4.9 e-mail thread
+after trimming e-mail boby and send out.
```
Comment (danrue) [inline docs/validate-stable-rc-builds.md]: "boby" ok that's just a funny typo :)

**Review by danrue — COMMENTED** [review]
Thanks @nareshkamboju. I left a bunch of suggestions. I think overall the content is good, but most of it needs to be more specific. Consider your audience to be a new employee that has to do your job, but doens't know where anything is. Hope that helps,
Dan

File: docs/validate-stable-rc-builds.md (hunk 1 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: Added required links and description.

File: docs/validate-stable-rc-builds.md (hunk 3 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: updated documents and addressed your comments.

File: docs/validate-stable-rc-builds.md (hunk 4 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md:39]: More examples have been added.

File: docs/validate-stable-rc-builds.md (hunk 5 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: I have provided links and examples.

File: docs/validate-stable-rc-builds.md (hunk 9 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: Provided detailed explanation with examples.

File: docs/validate-stable-rc-builds.md (hunk 10 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: Updated document.

File: docs/validate-stable-rc-builds.md (hunk 11 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: This section not need to discuss here.
so i have removed lava priority section.

File: docs/validate-stable-rc-builds.md (hunk 12 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md:138]: For easy reference I would like to keep list of devices, test names and number of tests.

File: docs/validate-stable-rc-builds.md (hunk 13 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: Typo fixed now.

**Comment by nareshkamboju** [comment]
I have updated the document with more details with examples.
Requesting team to review this document.

File: docs/validate-stable-rc-builds.md [hunk 14]
```diff
@@ -0,0 +1,237 @@
+# Stable rc review validation process
+Following steps to be performed when a stable rc review pushed on stable mailing
+list
+
+## Wait for stable rc review email
+Greg KH pushes on to stable rc trees and creates a tag
```
Comment (mrchapp) [inline docs/validate-stable-rc-builds.md]: Actually, no tag is created/published. (That would make things easier for lots of people.)

File: docs/validate-stable-rc-builds.md [hunk 15]
```diff
@@ -0,0 +1,237 @@
...
+then sends out email on stable ML with the list of patches included in that
+branch.
+
+stable-rc tree link:
+
+*https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux-stable-rc.git*
+
+stable-rc review email example link,
+
+*https://www.spinics.net/lists/stable/msg320559.html*
```
Comment (mrchapp) [inline docs/validate-stable-rc-builds.md]: That's a stable release. Maybe we can use this RC review request from today: https://lore.kernel.org/stable/20190808190453.582417307@linuxfoundation.org/T/#mda3573d436265842df547701dc61f9e236627d6c

File: docs/validate-stable-rc-builds.md [hunk 16]
```diff
@@ -0,0 +1,237 @@
...
+
+one must have to subscribe to the stable mailing list.
+You may check your inbox with given filters to identify the stable rc review
+for a given branch.
+
+## Build Test
+### Check for build is PASS
+open lkft.linaro.org website and check the build status
+From *https://lkft.linaro.org* you will re-directed to each branch builds for
+each device. If you see build status is in progress wait for the
```
Comment (mrchapp) [inline docs/validate-stable-rc-builds.md:34]: Or if there are builds queued.

Also, 4.4 has two jobs: one for HiKey and one for the rest.

Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md:34]: All 4.4 is also listed on lkft.linaro.org
so no need to specify about 4.4 hikey build

File: docs/validate-stable-rc-builds.md (hunk 15 above)
Comment (nareshkamboju) [reply docs/validate-stable-rc-builds.md]: yeah this one looks good.
I have updated this link in the document.
